src/EditorFeatures/Test2/Diagnostics/DiagnosticServiceTests.vb

```diff
@@ -1940,8 +1940,15 @@ class MyClass
                 ' Get cached project diagnostics.
                 Dim diagnostics = diagnosticService.GetCachedDiagnosticsAsync(workspace, project.Id).WaitAndGetResult(CancellationToken.None)
 
-                Assert.Equal(1, diagnostics.Count())
-                Assert.Equal(HiddenDiagnosticsCompilationAnalyzer.Descriptor.Id, diagnostics.Single().Id)
+                ' different behavior between v1 and v2. in v2. in v2, solution crawler never creates non-local hidden diagnostics.
```

Comment:
So I looked back at the original issue for which this test was added: https://github.com/dotnet/roslyn/issues/4068. It seems like 2 customers reported it and both of them report hidden diagnostics from compilation action. Also note that the definition of local is subtle even for open files: if you have analyzer that analyses a declaration in a file and reports a diagnostic outside that declaration in the same file, the diagnostic is classified as a non-local diagnostic. 

I think we really need a way for analyzers to tell don't run me on closed files to turn them off on closed files - unless we do that, we may have to follow @srivatsn suggestion and port the way v1 works - using IBuiltInAnalyzer.GetDiagnosticAnalysisCategory to skip running on closed files.